src/deco-subsurface.cpp

```diff
@@ -51,14 +51,17 @@ class simple_decoration_node_t : public wf::scene::node_t, public wf::pointer_in
     {
         if (auto view = _view.lock())
         {
-            int target_width  = width * scale;
-            int target_height = height * scale;
-
-            auto surface = theme.render_text(view->get_title(),
-                target_width, target_height, t_width, view->activated);
-            cairo_surface_upload_to_texture(surface, title_texture.tex);
-            cairo_surface_destroy(surface);
-            title_texture.current_text = view->get_title();
+            if (view->get_title() != title_texture.current_text)
```

Comment:
This needs to also be called on resize.

Reply:
.. and on focus change.

Reply:
I'm running with [this patch](https://termbin.com/a81m), seems to fix the issue.